client/extensions/woocommerce/app/order/order-refund-card.js

```diff
@@ -0,0 +1,196 @@
+import FormTextarea from 'components/forms/form-textarea';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import Notice from 'components/notice';
+import OrderDetailsTable from './order-details-table';
+import { sendRefund } from 'woocommerce/state/sites/orders/actions';
+import StoredCard from 'my-sites/upgrades/checkout/stored-card';
+
+class OrderRefundCard extends Component {
+	static propTypes = {
+		order: PropTypes.shape( {
+			currency: PropTypes.string.isRequired,
+			id: PropTypes.number.isRequired,
+			payment_method_title: PropTypes.string.isRequired,
+			refunds: PropTypes.array.isRequired,
+			status: PropTypes.string.isRequired,
+			total: PropTypes.string.isRequired,
+		} ),
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+			slug: PropTypes.string.isRequired,
+		} ),
+	}
+
+	state = {
+		errorMessage: false,
+		refundTotal: 0,
+		refundNote: '',
+		showDialog: false,
+	}
+
+	getRefundedTotal = ( order ) => {
+		return order.refunds.reduce( ( sum, i ) => sum + parseFloat( i.total ), 0 );
+	}
+
+	getRefundStatus = () => {
+		const { order, translate } = this.props;
+		let refundStatus = translate( 'Payment of %(total)s received via %(method)s', {
+			args: {
+				total: formatCurrency( order.total, order.currency ) || order.total,
+				method: order.payment_method_title,
+			}
+		} );
+
+		if ( 'refunded' === order.status ) {
+			refundStatus = translate( 'Payment of %(total)s has been refunded', {
+				args: {
+					total: formatCurrency( order.total, order.currency ) || order.total,
+				}
+			} );
+		} else if ( order.refunds.length ) {
+			const refund = this.getRefundedTotal( order );
+			refundStatus = translate( 'Payment of %(total)s has been partially refunded %(refund)s', {
+				args: {
+					total: formatCurrency( order.total, order.currency ) || order.total,
+					refund: formatCurrency( refund, order.currency ) || refund,
+				}
+			} );
+		}
+		return refundStatus;
+	}
+
+	toggleDialog = () => {
+		this.setState( {
+			errorMessage: false,
+			refundTotal: 0,
+			refundNote: '',
+			showDialog: ! this.state.showDialog,
+		} );
+	}
+
+	recalculateRefund = ( total ) => {
+		this.setState( { refundTotal: total } );
+	}
+
+	updateNote = ( event ) => {
+		this.setState( {
+			refundNote: event.target.value,
+		} );
+	}
+
+	sendRefund = () => {
+		const { order, site, translate } = this.props;
+		const maxRefund = parseFloat( order.total ) + this.getRefundedTotal( order );
+		if ( this.state.refundTotal > maxRefund ) {
+			this.setState( { errorMessage: translate( 'Refund must be less than order total.' ) } );
+			return;
+		} else if ( this.state.refundTotal <= 0 ) {
+			this.setState( { errorMessage: translate( 'Refund must be greater than zero.' ) } );
+			return;
+		}
+		this.toggleDialog();
+		const refundObj = {
+			amount: this.state.refundTotal + '', // API expects a string
+			reason: this.state.refundNote,
+			// api_refund
```

Comment:
This line will turn on/off automated refunds, depending on whether this payment method supports it